### ISSUE TITLE
External reference support for the JSON schema

### ISSUE THREAD

**Issue by nikhilsamdaniel** [ngx-formly/ngx-formly#3355]
I am trying to generate formly fields using FormlyJsonschema by passing the JSON schema file containing external references to other schema files. And I am getting an error saying the remote schema is not supported yet.

I am looking for a solution (i) that accepts external URLs in the schema reference or 
(ii) adds all the external referenced schema files and provides a combined resolved schema.

Currently, I am passing my schema using the below-mentioned way.
`this.formlyJsonschema.toFieldConfig(schema)`
Providing a sample schema below for the second option. Here, main.json is the schema in which an external reference to the base.json schema is mentioned.
```
base.json
{
  "$id": "base.json",  
  "type": "object",
  "properties": {
    "version": {
      "description": "version",
      "type": "string",
    }
  }
}
```
```
main.json
{
  "$id": "main.json",  
  "type": "object",
  "properties": {
    "version": {
        "$ref": "base.json#/properties/version"
    },
  }
}
```

If I am missing something, please correct me.

**Comment by kenisteward** [comment]
TLDR: it is in the works but not currently supported


External reference support currently doesn't exist. It has been something
that I've been working on for a while now but never have a proper solution
in comfortable sharing. Internally, we resolve all external dependencies
before we bind to formly.

I'd suggest doing that for now as the framework doesn't support it quite
yet. There's just quite a few things that need to be resolved

1. Properly handling id's
2. Do we resolve deps as we parse or before formly generates fields (  this
is important because that will change the return API of the Json schema
function )


There are a couple other things I'm sure I'm leaving out too posting this
on my phone.

**Comment by aitboudad** [comment]
PR welcomed!
